src/singpass-button/singpass-button.style.tsx

```diff
@@ -0,0 +1,131 @@
+import styled, { css } from "styled-components";
+import { ComponentLoadingSpinner } from "../shared/component-loading-spinner";
+import { Colour, MediaQuery, Motion, Spacing } from "../theme";
+import { ThemeButton } from "../theme/components/theme-helper";
+import { SingpassButtonStyleType } from "./types";
+
+export type MainButtonSize = "default" | "small" | "large";
+
+export interface MainStyleProps {
+    $buttonStyle: SingpassButtonStyleType;
+    $buttonSizeStyle?: MainButtonSize | undefined;
+    $singpassLogoOverride?: SingpassLogoOverride;
+}
+
+export interface SingpassLogoOverride {
+    height?: number;
+    translateY?: number;
+    paddingLeft?: number;
+}
+
+export const Main = styled.button<MainStyleProps>`
+    padding: ${Spacing["spacing-8"]} ${Spacing["spacing-16"]};
+    min-width: 4rem;
```

Comment:
remove this, I want to move away from hardcoding widths as I realised devs keep trying to override them anyway